payment/models/coupon.py

```diff
@@ -35,3 +37,22 @@ def get_effects_display(self):
             html_formatted_effects_display += f"{coupon_effect}<br>"
 
         return format_html(html_formatted_effects_display)
+
+    def biggest_discount_for(self, course, original_price):
```

Comment:
Should methods start with a verb for clarity, e.g. `calc_biggest_discount`?

Reply:
Thanks for thinking of and implementing this logic that only biggest discount applies

Reply:
sure, will make the change!

Reply:
actually, don't think this is necessary. i think it's ok to leave the naming as is since the method actually returns the discount.